pymc_marketing/mmm/risk_assessment.py

```diff
@@ -0,0 +1,461 @@
+        return mean - alpha * tail_metric(samples, budgets)
+
+    return _mean_tightness_score
+
+
+def _calculate_roas_distribution_for_allocation(
+    samples: np.ndarray, budgets: np.ndarray
+) -> np.ndarray:
+    """Calculate the ROAS distribution for a given total budget."""
+    return samples / np.sum(budgets)
+
+
+def value_at_risk(confidence_level: float = 0.95) -> ObjectiveFunction:
+    R"""
+    Calculate the Value at Risk (VaR) at a specified confidence level.
+
+    VaR estimates the potential loss in value of an asset or portfolio over a defined period
+    for a given confidence interval. It is a standard measure used in risk management to
+    assess the risk of loss on a specific portfolio of financial assets.
+
+    The Value at Risk (VaR) is calculated as:
+
+        .. math::
+            VaR = \mu - Q_{(1 - \alpha)}
+
+    where:
+        - :math:`\mu` is the mean of the sample returns.
+        - :math:`Q_{(1 - \alpha)}` is the quantile at the specified confidence level.
+
+    Parameters
+    ----------
+    confidence_level : float, optional
+        Confidence level for VaR (default is 0.95).
+
+    Returns
+    -------
+    ObjectiveFunction
+        A function that calculates the VaR value at the specified confidence level given samples and budgets.
+
+    Raises
+    ------
+    ValueError
+        If confidence_level is not between 0 and 1.
+
+    References
+    ----------
+    - Jorion, P. (2006). Value at Risk: The New Benchmark for Managing Financial Risk.
+    """
+
+    def _value_at_risk(samples: np.ndarray, budgets: np.ndarray) -> float:
+        if not 0 < confidence_level < 1:
+            raise ValueError("Confidence level must be between 0 and 1.")
+        return np.percentile(samples, (1 - confidence_level) * 100)
+
+    return _value_at_risk
+
+
+def conditional_value_at_risk(confidence_level: float = 0.95) -> ObjectiveFunction:
+    R"""
+    Calculate the Conditional Value at Risk (CVaR) at a specified confidence level.
+
+    CVaR, also known as Expected Shortfall, measures the average loss exceeding the VaR
+    at a given confidence level, providing insight into the tail risk of the distribution.
+
+    The Conditional Value at Risk (CVaR) is calculated as:
+
+        .. math::
+            CVaR = \mathbb{E}[X \mid X \leq VaR]
+
+    where :math:`X` represents the loss distribution, and :math:`VaR` is the Value at Risk
+    at the specified confidence level. CVaR provides a more comprehensive view of the risk
+    associated with extreme losses beyond the VaR.
+
+    Parameters
+    ----------
+    confidence_level : float, optional
+        Confidence level for CVaR (default is 0.95).
+
+    Returns
+    -------
+    ObjectiveFunction
+        A function that calculates the CVaR value at the specified confidence level given samples and budgets.
+
+    Raises
+    ------
+    ValueError
+        If confidence_level is not between 0 and 1.
+        If no samples fall below the VaR threshold.
```

Comment:
This technically isn't raised by this function but by the inner function so I would remove